detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/BitbakeSession.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * detectable
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectables.bitbake;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detectable.detectable.executable.ExecutableOutput;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableRunner;
+import com.synopsys.integration.detectable.detectable.executable.ExecutableRunnerException;
+import com.synopsys.integration.detectable.detectable.file.FileFinder;
+import com.synopsys.integration.detectable.detectables.bitbake.model.BitbakeFileType;
+import com.synopsys.integration.detectable.detectables.bitbake.model.BitbakeRecipe;
+import com.synopsys.integration.detectable.detectables.bitbake.model.BitbakeResult;
+import com.synopsys.integration.detectable.detectables.bitbake.parse.BitbakeLayersParser;
+import com.synopsys.integration.detectable.detectables.bitbake.parse.BitbakeRecipesParser;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.IntLogger;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+public class BitbakeSession {
+    private final IntLogger logger = new Slf4jIntLogger(LoggerFactory.getLogger(this.getClass()));
+
+    private final FileFinder fileFinder;
+    private final ExecutableRunner executableRunner;
+    private final BitbakeLayersParser bitbakeLayersParser;
+    private final BitbakeRecipesParser bitbakeRecipesParser;
+    private final File outputDirectory;
+    private final File buildEnvScript;
+    private final String[] sourceArguments;
+    private final File bashExecutable;
+
+    public BitbakeSession(final FileFinder fileFinder, final ExecutableRunner executableRunner, final BitbakeLayersParser bitbakeLayersParser,
```

Comment:
I like to split data from work. I'd prefer a BitbakeSession and maybe a BitbakeRunner or something similar.